File: mobile/src/main/kotlin/de/scout/fireplace/inject/ActivityBindingModule.java
package de.scout.fireplace.inject;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import de.scout.fireplace.home.HomeActivity;
import de.scout.fireplace.home.HomeModule;
import de.scout.fireplace.login.LoginActivity;
import de.scout.fireplace.login.LoginModule;
import de.scout.fireplace.settings.SettingsActivity;
import de.scout.fireplace.settings.SettingsModule;

@Module
public interface ActivityBindingModule {

  @ContributesAndroidInjector(modules = HomeModule.class)
  HomeActivity homeActivity();

  @ContributesAndroidInjector(modules = LoginModule.class)
  LoginActivity loginActivity();

  @ContributesAndroidInjector(modules = SettingsModule.class)
  SettingsActivity preferenceActivity();
}

File: feature/src/main/kotlin/de/scout/fireplace/feature/inject/ActivityBindingModule.kt
package de.scout.fireplace.feature.inject

import dagger.Module
import dagger.android.ContributesAndroidInjector
import de.scout.fireplace.feature.home.HomeActivity
import de.scout.fireplace.feature.home.HomeModule
import de.scout.fireplace.feature.login.LoginActivity
import de.scout.fireplace.feature.login.LoginModule
import de.scout.fireplace.feature.settings.SettingsActivity
import de.scout.fireplace.feature.settings.SettingsModule

@Module
internal interface ActivityBindingModule {

  @ActivityScope
  @ContributesAndroidInjector(modules = arrayOf(HomeModule::class))
  fun homeActivity(): HomeActivity

  @ActivityScope
  @ContributesAndroidInjector(modules = arrayOf(LoginModule::class))
  fun loginActivity(): LoginActivity

  @ActivityScope
  @ContributesAndroidInjector(modules = arrayOf(SettingsModule::class))
  fun preferenceActivity(): SettingsActivity
}
